code/RobustLight2.dm

```diff
@@ -211,6 +211,9 @@ datum/light
 				if (SHOULD_QUEUE)
 					light_update_queue.queue(src)
 					dirty_flags |= D_COLOR
+					r_des = red
+					g_des = green
+					b_des = blue
```

Comment:
As I understand changing datum variables is Bad, but I don't know how else to fix the light update bug. If I understand right this would set all simultaneously queued lights the same colour, probably the wrong one if there's more than one light queued. In my testing I've never had anything go wrong though, so possibly that scenario doesn't happen in practice. Still, not ideal.

(For reference, r_des and the others are only used for setting the colour on queued lights, so this doesn't break other things. They're normally all set to 1.)

Reply:
I think this is correct and it missing was an oversight. (Unless I misunderstood how RL handles queues.)